frontend/pages/model/[modelId].tsx

```diff
@@ -17,24 +18,30 @@ export default function Model() {
   const router = useRouter()
   const { modelId }: { modelId?: string } = router.query
   const { model, isModelLoading, isModelError } = useGetModel(modelId)
+  const { currentUser, isCurrentUserLoading, isCurrentUserError } = useGetCurrentUser()
   const { uiConfig, isUiConfigLoading, isUiConfigError } = useGetUiConfig()
 
+  const roles = useMemo(
```

Comment:
It's a bit verbose, but it might help to clarify what roles these are. We also have the concept of modelroles, which is an array of objects that may not be related to the current user. Consider changing all of the `roles` in this PR to be `currentUserRoles` or similar.

Also, when working with arrays it tends to make logic easier down the line if you always return an array. So you could do:
```
  const currentUserRoles = useMemo(
    () =>
      model?.collaborators.find((collaborator) => collaborator.entity.split(':')[1] === currentUser?.dn)?.roles || [],
    [model, currentUser],
  )
```
You could then remove the `roles` check in the `tabs` memo hook below.